File: Java/SingleLinkedList.java

public class SingleLinkedList<T> implements LinkedList<T> {

    private Node last = null;

    private class Node {
        private T item;
        private Node previous;
    }

    public boolean isEmpty() {
        return last == null;
    }

    public void push(final T item) {
        Node oldLast = last;
        last = new Node();
        last.item = item;
        last.previous = oldLast;
    }

    public T pop() {
        T item = null;
        if (!isEmpty()) {
            item = last.item;
            last = last.previous;
        }
        return item;
    }
}

File: CSharp/SingleLinkedList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LinkedList
{
    class SingleLinkedList<T>
    {
        private Node last = null;

        private class Node
        {
            public T item;
            public Node previous;
        }

        public bool isEmpty()
        {
            return last == null;
        }

        public void addNode(T item)
        {
            Node oldLast = last;
            last = new Node();
            last.item = item;
            last.previous = oldLast;
        }

        public List<T> printList()
        {
            List<T> res = new List<T>();
            T item = default(T);
            while (!isEmpty())
            {
                item = last.item;
                last = last.previous;
                res.Add(item);
            }
            return res;
        }
    }
}
